### DISCUSSION TITLE
Support for multiple network interfaces

### DISCUSSION THREAD

**Discussion by debsahu** [hassio-addons/addon-adguard-home#192] [Ideas]
# Problem/Motivation

I have two network interfaces, enp6s0 and enp5s0. Both get a ipv4 & ipv6 network addresses. Hassio randomly picks one of the two interfaces and passes it to AdGaurd. With the new update, AdGaurd 4.1.2 only binds to ipv4/v6 addresses of the randomly chosen interfaces and not all network interface.

## Expected behavior

Bind to both physical network interfaces just like v3.x.x

## Actual behavior

Binds to one of the two network interfaces.

## Steps to reproduce

- Install more than one physical network interface
- Start AdGaurd v4.1.2

## Proposed changes

Wish I knew how to add back this feature. Calling in experts in our community to propose changes.

**Comment by frenck** [comment]
Converted to a discussion, as this is a feature request, not an issue.

**Comment by debsahu** [comment]
Just for reference, this feature was present in older v3.x.x versions of add-on. So don’t know if this qualifies as feature request as currently this version v4.x.x strips this functionality. Is there a way to read from config file?

**Reply by frenck** [reply]
4.x listed this as a breaking change. It was removed deliberately and it not a bug.
You are asking support for multiple interfaces, 3.x didn't do that either, as 3.x bound to ALL it could find (and causing problems because of that).

**Reply by debsahu** [reply]
Makes sense, some != all. Right now the add-on reads what hassio passes as default interface. Btw thanks for all that you do. Would you point me to anywhere I can read up on adding such functionality and if I succeed, I’ll create a PR.

**Comment by pickeld** [comment]
i probably missing something, this addon listen to port 53 on the loopback interface, therefore any external device wont be able to reach it.
how is this a feature?

**Reply by debsahu** [reply]
I’m guessing the host os Fedora/Ubuntu/Debian will be able to reach the internet.

**Reply by pickeld** [reply]
Thia is a network solution not a single client solution. It must listen on the right segments or 0.0.0.0

**Reply by frenck** [reply]
> i probably missing something, this addon listen to port 53 on the loopback interface

Yes, you are missing something. It asks the Supervisor for the network interfaces and uses those. The Supervisor manages the network on Home Assistant Supervised & OS-based systems.

>  It must listen on the right segments or 0.0.0.0

No, absolutely not. That is wrong for DNS and will cause problems with e.g., systemd-resolved, which is quite standard.

**Reply by pickeld** [reply]
hey @frenck , who routes the traffic from the host to the container?

host:
> root@hass:~# netstat -lnp | grep 53
> tcp        0      0 127.0.0.1:53            0.0.0.0:*               LISTEN      11695/./AdGuardHome
> udp        0      0 127.0.0.1:53            0.0.0.0:*                           11695/./AdGuardHome

client
> C:\Users\x>nslookup cnn.com 192.168.127.250
> DNS request timed out.
>     timeout was 2 seconds.
> Server:  UnKnown
> Address:  192.168.127.250
> 
> DNS request timed out.
>     timeout was 2 seconds.
> DNS request timed out.
>     timeout was 2 seconds.
> DNS request timed out.
>     timeout was 2 seconds.
> DNS request timed out.
>     timeout was 2 seconds.
> *** Request to UnKnown timed-out

also, what if i uses multiple vlans which the supervisor does not familiar with?

**Reply by frenck** [reply]
> hey @frenck , who routes the traffic from the host to the container?

Nobody, it runs on the host network.

> also, what if i uses multiple vlans which the supervisor does not familiar with?

That isn't supported by the Home Assistant project, thus not a concern for add-ons either.
